core/shared/src/test/scala/fs2/StreamSuite.scala

```diff
@@ -1024,6 +1024,11 @@ class StreamSuite extends Fs2Suite {
     assert(compileErrors("Stream.eval(IO(1)).through(p)").nonEmpty)
   }
 
+  test("Unintended unchunking when using Monad instance of Stream") {
```

Comment:
you have only tested StreamLowPriority instance. I think that duplicating this test and adding `.covary[IO]` before `.through` (and `.compile` before `.toList`) will be sufficient :)

Reply:
Thanks! I will do that.